src/test/e2e/api/admin/project/project.api.tokens.e2e.test.ts

```diff
@@ -0,0 +1,115 @@
+import dbInit, { ITestDb } from '../../../helpers/database-init';
+import {
+    IUnleashTest,
+    setupAppWithCustomConfig,
+} from '../../../helpers/test-helper';
+import getLogger from '../../../../fixtures/no-logger';
+import { ApiTokenType } from '../../../../../lib/types/models/api-token';
+
+let app: IUnleashTest;
+let db: ITestDb;
+
+beforeAll(async () => {
+    db = await dbInit('project_api_tokens_serial', getLogger);
+    app = await setupAppWithCustomConfig(db.stores, {
+        experimental: {
+            flags: {
+                strictSchemaValidation: true,
+            },
+        },
+    });
+});
+
+afterEach(async () => {
+    await db.stores.apiTokenStore.deleteAll();
+});
+
+afterAll(async () => {
+    await app.destroy();
+    await db.destroy();
+});
+
+test('Returns empty list of tokens', async () => {
+    return app.request
+        .get('/api/admin/projects/default/api-tokens')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((res) => {
+            expect(res.body.tokens.length).toBe(0);
+        });
+});
+test('Returns list of tokens', async () => {
+    const tokenSecret = 'random-secret';
+
+    await db.stores.apiTokenStore.insert({
+        username: 'test',
+        secret: tokenSecret,
+        type: ApiTokenType.CLIENT,
+        environment: 'default',
+        projects: ['default'],
+    });
+    return app.request
+        .get('/api/admin/projects/default/api-tokens')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((res) => {
+            expect(res.body.tokens.length).toBe(1);
+            expect(res.body.tokens[0].secret).toBe(tokenSecret);
+        });
+});
+
+test('Returns 404 when given non-existant projectId', async () => {
+    return app.request
+        .get('/api/admin/projects/wrong/api-tokens')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((res) => {
+            expect(res.body.tokens.length).toBe(0);
+        });
+});
+
+test('fails to create new client token when given wrong project', async () => {
+    return app.request
+        .post('/api/admin/projects/wrong/api-tokens')
+        .send({
+            username: 'default-client',
+            type: 'client',
+            projects: ['wrong'],
+            environment: 'default',
+        })
+        .set('Content-Type', 'application/json')
+        .expect(400);
```

Comment:
Ooh, should this be a 400 or 404? I don't have the answer, but it feels like if the URL is wrong, then it's a 404, but if the request payload is bad, then that's a 400 to me. What do **you** think?

Reply:
It feels like a 404 to me - I'll add a task to fix the endpoint